logstash-core/src/main/java/org/logstash/secret/store/SecureConfig.java

```diff
@@ -0,0 +1,78 @@
+package org.logstash.secret.store;
+
+import java.nio.CharBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A String -> char[] container that holds a referenced char[] obfuscated in memory and allows for easy clearing all values from memory.
```

Comment:
I'm imaginging the purpose of this class will change very quickly if we have multiple SecureStore implementations. For example, if we have a networked one (talking to Vault or Elasticsearch?), we'll need connection details in addition to credentials. 

No change required today, but do you envision this SecureConfig map-like construct being used for all of this (especially as all values are char[]) ?

Reply:
I already kinda have that issue since I need to push the `keystore.path` through this class.  While it is a bit awkward to push non-sensitive stuff through through this config (due to char[]), it is isn't too hard to work with. 

 For example, from Ruby:
```
secureConfig.add("keystore.path", LogStash::SETTINGS.get_setting("path.keystore").value.chars)
```

 Imo, this make more sense then having two configurations, or adding an unsecure (String based) part to the secure config.